components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -4679,6 +4655,10 @@ public void deleteAPI(APIIdentifier apiId) throws AppManagementException {
 			prepStmt.setInt(1, id);
 			prepStmt.execute();
 
+			prepStmt = connection.prepareStatement(deleteConsumerQuery);
+			prepStmt.setString(1, apiId.getApiName() + "-" + apiId.getVersion());
+			prepStmt.execute();
```

Comment:
Yes, there is an issue not closing the statements properly. The issue was there in the original code, but I overlooked it. I will fix this in all relevant locations in this file.